core/src/test/java/org/apache/calcite/avatica/remote/looker/LookerDriverTest.java

```diff
@@ -163,6 +181,84 @@ public void resultsCanBeParsedIntoResultSet() throws SQLException {
+    // so have been placed in the LookerTestCommon class to make this test easier to read.
+    Driver driver = new StubbedLookerDriver().withStubbedResponse(LookerTestCommon.stubbedListMeasuresSignature,
+        LookerTestCommon.stubbedListMeasureResponse);
+    Connection connection = driver.connect(LookerTestCommon.getUrl(),
+        LookerTestCommon.getBaseProps());
+
+    ResultSet test = connection.createStatement().executeQuery(TEST_LIST_MEASURE_SQL);
+    ResultSetMetaData rsMetaData = test.getMetaData();
+
+    // verify column types
+    Assert.assertThat(rsMetaData.getColumnCount(), is(3));
+
+    // YEAR
+    Assert.assertThat(rsMetaData.getColumnType(1), is(Types.INTEGER));
+    Assert.assertThat(rsMetaData.getColumnName(1), is("users.created_year"));
+
+    // Names
+    Assert.assertThat(rsMetaData.getColumnType(2), is(Types.ARRAY));
+    Assert.assertThat(rsMetaData.getColumnName(2), is("names"));
+
+    // Ages
+    Assert.assertThat(rsMetaData.getColumnType(3), is(Types.ARRAY));
+    Assert.assertThat(rsMetaData.getColumnName(3), is("ages"));
+
+    // verify every row can be fetched with the appropriate getter method
+    while (test.next()) {
+      assertNotNull(test.getInt(1));
+      assertNotNull(test.getArray(2));
+      assertNotNull(test.getArray(3));
+    }
+  }
+
+  @Test
+  public void canProcessAllArrayTypes() throws SQLException {
+    // Set up the driver with some pre-recorded responses from Looker. These are large JSON strings
+    // so have been placed in the LookerTestCommon class to make this test easier to read.
+    Driver driver = new StubbedLookerDriver().withStubbedResponse(LookerTestCommon.stubbedArrayPrimitivesSig,
+        LookerTestCommon.stubbedArrayPrimitivesResponse);
+    Connection connection = driver.connect(LookerTestCommon.getUrl(),
+        LookerTestCommon.getBaseProps());
+
+    ResultSet test = connection.createStatement().executeQuery(TEST_ARRAY_PRIMITIVES_SQL);
+    ResultSetMetaData rsMetaData = test.getMetaData();
+    int columnCount = rsMetaData.getColumnCount();
+
+    // Verify we got our arrays
+    for (int i = 2; i <= columnCount; i++) {
+      Assert.assertThat(rsMetaData.getColumnType(i), is(Types.ARRAY));
+    }
+
+     while (test.next()) {
+       Object outputArray = test.getArray(2).getArray();
+       Assert.assertThat(new int[]{1,2,3}, is(equalTo(outputArray)));
+
+       outputArray = test.getArray(3).getArray();
+       Assert.assertThat(
+           new BigDecimal[]{BigDecimal.valueOf(1.1), BigDecimal.valueOf(2.2), BigDecimal.valueOf(3.3)},
+           is(equalTo(outputArray))
+       );
+
+       outputArray = test.getArray(4).getArray();
+       Assert.assertThat(
+           new String[]{"this", "and", "that"},
+           is(equalTo(outputArray))
+       );
+
+       outputArray = test.getArray(5).getArray();
+       Assert.assertThat(
+           new boolean[]{true, false, false},
```

Comment:
should we intiialize the expected outputs like you did below with `expectedOutputValues`? Might make assertion easier to read.